File: leisure-games-webapp-backend/src/main/java/com/example/laisuregameswebappbackend/model/Game.java
package com.example.laisuregameswebappbackend.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name="game")
@Data
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "name")
    private String name;
    @Column(name = "fame")
    private String fame;
    @Column(name="short_desc")
    private String shortDesc;
    @Column(name = "full_desc")
    private String fullDesc;
    @Column(name = "image_link")
    private String imageLink;
    @Column(name="category")
    private String category;
}

File: leisure-games-webapp-frontend/src/app/game.ts
export class Game {
    id:number;
    name:string;
    fame:string;
    shortDesc:string;
    fullDesc:string;
    imageLink:string;
    category:string;

    constructor(id?: number, name?: string, fame?: string, shortDesc?:string, fullDesc?: string, imageLink?: string, category?: string) {
        this.id = id || 0;
        this.name = name || '';
        this.fame = fame || '';
        this.shortDesc = shortDesc || '';
        this.fullDesc = fullDesc || '';
        this.imageLink = imageLink || '';
        this.category = category || '';
    }
}
